src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -666,6 +669,70 @@ def create_application(client, display_name, homepage=None, identifier_uris=None
 
     return result
 
+def list_granted_application(cmd, identifier):
+    graph_client = _graph_client_factory(cmd.cli_ctx)
+
+    # Get the Service principal ObjectId for the client app
+    sps = list(list_sps(graph_client.service_principals, appid=identifier))
+
+    if not sps:
+        raise CLIError("'--id'" + identifier + " does not exist on this directory")
+    client_sp_object_id = sps[0].object_id
+
+    # Get the OAuth2 permissions client app
+    permissions = graph_client.oauth2.get(
+        filter="clientId eq '{}'".format(client_sp_object_id))
+
+    return permissions.additional_properties['value']
+
+def grant_application(cmd, identifier, appid, expires='1'):
+    graph_client = _graph_client_factory(cmd.cli_ctx)
+
+    # Get the Service principal ObjectId for the client app
+    sps = list(list_sps(graph_client.service_principals, appid=identifier))
```

Comment:
You can just use `_resolve_service_principal` which returns the object id you need.

Reply:
Implemented and extended to query by appId